app/presenters/tree_node/menu/item.rb

```diff
@@ -0,0 +1,39 @@
+module TreeNode
+  module Menu
+    class Item < TreeNode::Menu::Node
+      set_attribute(:key)     { "#{@options[:node_id_prefix]}__#{@object.feature}" }
+      set_attribute(:title)   { _(details[:name]) }
+      set_attribute(:tooltip) { _(details[:description]) || _(details[:name]) }
+
+      def select_state
+        self_selected? || select_by_children_selected
+      end
+
+      def children
```

Comment:
I think this may have been possible when I was building the whole tree with hashes, but now that I'm building them _The One True TreeBuilder Way_, the select state of the node must be determined when it's built. This requires knowing the select states of all the children in this class. In other words, the TreeBuilder doesn't have the knowledge necessary to set the select states correctly.